File: Java Codes/New Batch/PalindromeFunction.java
import java.util.Scanner;

class PalindromeFunction{
    public static int reverse(int num){
        int rev = 0;
        while(num>0){
            rev = rev*10 + (num%10);
            num /= 10;
        }
        return rev;
    }
    public static boolean isPalindrome(int num){
        if(num>=0 && num<=9){
            return true;
        }
        if(reverse(num)==num){
            return true;
        }
        return false;
    }
    public static void main(String[] args) {
        int num;
        System.out.println("Enter a number: ");
        Scanner sc = new Scanner(System.in);
        num = sc.nextInt();
        System.out.println(isPalindrome(num));
    }
}

File: C++ program/PalindromeFunction.cpp
#include<iostream>
using namespace std;
int reverse(int p){
    int rev = 0;
    int ldig;
    while(p>0){
        ldig = p%10;
        rev = rev*10 + ldig;
        p = p/10;
    }
    return rev;
}
bool isPalindrome(int num){
    if(num>=0 and num<=9){
        return true;
    }
    // if(num==reverse(num)){
    //     return true;
    // }
    // else{
    //     return false;
    // }
    return (num==reverse(num));
}
int main(){
    int num;
    cout<<"Enter a number: ";
    cin>>num;
    if(isPalindrome(num)){
        cout<<"Palindrome number";
    }
    else{
        cout<<"Non Palindrome number";
    }
    return 0;
}